Python: Getting Started ASGI (#7802)
getsentry/sentry-docs: src/platforms/python/common/configuration/integrations/asgi.mdx

import sentry_sdk
from sentry_sdk.integrations.asgi import SentryAsgiMiddleware

from my_asgi_app import app

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",

    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    traces_sample_rate=1.0,
)

app = SentryAsgiMiddleware(app)
```

The middleware supports both ASGI 2 and ASGI 3 transparently.

## Verify

> Trigger an error on purpose and see as it shows up in Sentry.

Reviewer: ```suggestion
Trigger an error in your code and see it show up in Sentry.
```

```python
sentry_sdk.init(...) # same as above

def app(scope):
    async def get_body():
        return f"The number is: {1/0}" # raises an error!

    async def asgi(receive, send):
        await send(
            {
                "type": "http.response.start",
                "status": 200,
                "headers": [[b"content-type", b"text/plain"]],
            }
        )
        await send({"type": "http.response.body", "body": await get_body()})

    return asgi

app = SentryAsgiMiddleware(app)
```

Run your ASGI app with unicorn:

```bash
uvicorn main:app --port 8000
```

> Point your browser to [http://localhost:8000](http://localhost:8000) and the error will be triggered and sent to Sentry.

Reviewer: ```suggestionwhi
Point your browser to [http://localhost:8000](http://localhost:8000) to trigger the error which is then sent to Sentry.
```

> Additionally a transaction will show up in the "Performance" section on Sentry.

Reviewer: ```suggestion
Additionally, a transaction will show up in the "Performance" section on Sentry.
```
